hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/TypedTable.java

```diff
@@ -69,8 +94,40 @@ public boolean isEmpty() throws IOException {
     return rawTable.isEmpty();
   }
 
+  /**
+   * Returns the value mapped to the given key in byte array or returns null
+   * if the key is not found.
+   *
+   * First it will check from cache, if it has entry return the value
+   * otherwise, get from the RocksDB table.
+   *
+   * @param key metadata key
+   * @return VALUE
+   * @throws IOException
+   */
   @Override
   public VALUE get(KEY key) throws IOException {
+    // Here the metadata lock will guarantee that cache is not updated for same
+    // key during get key.
+    if (cache != null) {
+      CacheValue<VALUE> cacheValue = cache.get(new CacheKey<>(key));
+      if (cacheValue == null) {
+        return getFromTable(key);
+      } else {
+        // Doing this because, if the Cache Value Last operation is deleted
+        // means it will eventually removed from DB. So, we should return null.
+        if (cacheValue.getLastOperation() != CacheValue.OperationType.DELETED) {
+          return cacheValue.getValue();
+        } else {
+          return null;
```

Comment:
Instead of making delete operation a special case, in case of delete we can just push a null value into the cache. So it will automatically return null.

Reply:
Yes, we can pass null value to CacheValue actual value. This will work. 
Done.

Reply:
Thanks. Also instead of null we can pass in Optional.absent(), same thing but makes it very clear that the value can be missing.

Reply:
Done.